luna/featurevis/objectives.py

```diff
@@ -0,0 +1,86 @@
+import tensorflow as tf
+
+class FilterObjective(object):
+    """ Feature visualization of a given channel in a layer of a model.
+        Computes the mean activation of the specified channel.
+    """
+        
+    def __init__(self, model, layer, filter_index, regularization=None):
+        """ Feature visualization of a given channel in a layer of a model.
+            Computes the mean activation of the channel.
+            Args: 
+                model (object): the model to be used for the feature visualization.
+                layer (string): the name of the layer to be used in the visualization.
+                filter_index (number): the index of the filter to be visualized.
+                regularization (function): customized regularizers to be applied. Defaults to None.
+        """
+        self.model = get_feature_extractor(model, layer)
+        self.filter_index = filter_index
+        self.regularization = regularization
+
```

Comment:
Since pylint does not like classes with just one public method, it could be useful to implement a standard __repr__ function:

```suggestion
def __repr__(self) -> str:
        return f"FilterObjective({self.model},\
                   {self.filter_index}, {self.regularization})"
```

Alternatively, we could just deactivate the pylint error at the beginning of the class with:

" # pylint: disable=R0903 "